### PR TITLE
Add a new Resource type to schema authz

### PR DESCRIPTION
**Pull request by dougwettlaufer**

### CODE REVIEW

File: authnz/src/main/java/io/stargate/auth/Resource.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+public enum Resource {
+  KEYSPACE,
+  TABLE,
+  TYPE,
+  TRIGGER,
+  FUNCTION,
+  AGGREGATE,
+  VIEW,
+  INDEX
```
Comment (dimas-b) [inline authnz/src/main/java/io/stargate/auth/Resource.java:11]: This looks very similar to our existing `io.stargate.auth.entity.ResourceKind`... Could it be enhanced and reused?

Comment (dimas-b) [reply authnz/src/main/java/io/stargate/auth/Resource.java:11]: Also `AuthorizedResource` might be relevant too.

File: graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/ddl/fetchers/KeyspaceDto.java [hunk 2]
```diff
@@ -88,7 +89,8 @@ private boolean isAuthorized(Table table) {
           authenticationSubject,
           Collections.singletonList(keyspace.name()),
           Collections.singletonList(table.name()),
-          SourceAPI.GRAPHQL);
+          SourceAPI.GRAPHQL,
+          ResourceKind.KEYSPACE);
```
Comment (dimas-b) [inline graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/ddl/fetchers/KeyspaceDto.java:93]: `ResourceKind.TABLE`?

Comment (dougwettlaufer) [reply graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/ddl/fetchers/KeyspaceDto.java]: Fixed. I got overzealous with the copy paste :)

**Review by dimas-b — APPROVED** [review]

**Pull request merged**